### PR TITLE
take two warnings. `RCTDatePickerIOS.propTypes` don't pass the verify…

### PR DESCRIPTION
**Pull request by epooren**
only modify a little bit to resolve an issue #6264

### CODE REVIEW

**Comment by javache** [comment]
@facebook-github-bot shipit

File: Libraries/Components/DatePicker/DatePickerIOS.ios.js [hunk 1]
```diff
@@ -123,7 +123,7 @@ var DatePickerIOS = React.createClass({
         <RCTDatePickerIOS
           ref={ picker => this._picker = picker }
           style={styles.datePickerIOS}
-          date={props.date.getTime()}
+          date={props.date}
```
Comment (javache) [inline Libraries/Components/DatePicker/DatePickerIOS.ios.js]: According to @nicklockwood this is actually wrong, React just verifies the wrong prop types somehow.

**Comment by nicklockwood** [comment]
I don't understand how this can work. You're now passing a JavaScript Date object directly to the native component, which means the Date is going to be converted to a JSON string and passed over the bridge.

I would have thought that passing a Date directly into JSON.stringify() would be an error.

File: Libraries/Components/DatePicker/DatePickerIOS.ios.js [hunk 2]
```diff
@@ -133,7 +133,7 @@ var DatePickerIOS = React.createClass({
           mode={props.mode}
           minuteInterval={props.minuteInterval}
           timeZoneOffsetInMinutes={props.timeZoneOffsetInMinutes}
-          onChange={this._onChange}
+          onDateChange={this._onChange}
```
Comment (nicklockwood) [inline Libraries/Components/DatePicker/DatePickerIOS.ios.js]: I'm pretty sure this change is also wrong - there is no onDateChange event exported by the native component.

**Comment by ide** [comment]
@nicklockwood fyi Date objects in JS implement a method called `toJSON`, which `JSON.stringify` calls if that method is defined. But.... this probably still breaks things because Date.toJSON returns a date string instead of unixtime...

**Comment by epooren** [comment]
@nicklockwood now, i sure my changes were wrong, but why this changes can work?
i changed thie according to `DatePickerIOS.propTypes` and `var RCTDatePickerIOS = requireNativeComponent('RCTDatePicker', DatePickerIOS, {})`

**Comment by nicklockwood** [comment]
@ide ah, that explains it. The [RCTConvert NSDate:] function accepts dates in string or numeric timestamp form.

It's much less efficient to use string -> date conversion though. I think we can probably find a better solution. Maybe add the date -> timestamp conversion automatically inside requireNativeComponent, as we do for colors, imageSource, etc.

**Comment by epooren** [comment]
@nicklockwood how to fix the `onDateChange`?  i think it should be configed `reactOnly` at `requireNativeComponent`

**Comment by epooren** [comment]
close

**Pull request closed without merging**